zerver/views/users.py

```diff
@@ -109,6 +110,23 @@ def update_user_backend(request, user_profile, email,
 
     return json_success()
 
+@has_request_variables
+def update_buddy_list(request, user_profile, user_email=REQ(validator=check_string),
+                      buddy_email=REQ(validator=check_string), should_add=REQ(validator=check_string)):
+    # type: (HttpRequest, UserProfile, str, str, str) -> HttpResponse
+    try:
+        user_profile = get_user_profile_by_email(user_email)
+    except UserProfile.DoesNotExist:
+        return json_error(_('No such user with email id ' + user_email))
+
+    try:
+        buddy_profile = get_user_profile_by_email(buddy_email)
```

Comment:
Let's reference buddy users by user ID, not email (we're in general migrating to only use IDs, not emails, to reference users to simplify email address changes).